### PR TITLE
Add basic implementation of LOI cards

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards #1153 

* Adds a recycler view to the MapContainerFragment
* Displays a card item for all LOIs in the active survey.
* On switching the survey, the cards also get updated.
* Only basic information (jobName, loiCaption, etc.) is included

@gino-m PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapContainerFragment.kt [hunk 1]
```diff
@@ -105,6 +110,23 @@ class MapContainerFragment @Inject constructor(private var mapsRepository: MapsR
       .getZoomThresholdCrossed()
       .`as`(RxAutoDispose.autoDisposable(this))
       .subscribe { onZoomThresholdCrossed() }
+    mapContainerViewModel.locationsOfInterest.observe(this) { onLoiChange(it) }
+  }
+
+  private fun onLoiChange(it: ImmutableSet<LocationOfInterest>) {
+    val list =
+      it
+        .stream()
```
Comment (JSunde) [inline ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapContainerFragment.kt:119]: Is this `stream()` necessary?

File: ground/src/main/res/layout/loi_card_item.xml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  <androidx.constraintlayout.widget.ConstraintLayout
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:orientation="vertical">
+    <TextView
+      android:id="@+id/jobName"
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:layout_gravity="center"
+      android:layout_marginStart="10dp"
```
Comment (JSunde) [inline ground/src/main/res/layout/loi_card_item.xml:35]: Would it make sense to pull this 10dp out into a dimension and reference it in all the locations 10dp is used in this file?

**Review by JSunde — APPROVED** [review]

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapContainerFragment.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapContainerFragment.kt]: Removed

File: ground/src/main/res/layout/loi_card_item.xml (hunk 2 above)
Comment (shobhitagarwal1612) [reply ground/src/main/res/layout/loi_card_item.xml]: Done

**Pull request merged**